src/main/java/de/tum/in/www1/artemis/service/connectors/hades/HadesCITriggerService.java

```diff
@@ -0,0 +1,173 @@
+            var job = createJob(participation);
+            postJob(job);
+        }
+        catch (JsonProcessingException e) {
+            log.error("Error creating the build job for participation {}", participation.getId(), e);
+        }
+    }
+
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation, String commitHash) throws ContinuousIntegrationException {
+        log.warn("Triggering builds with a commit hash is not supported for Hades. Triggering build without commit hash.");
+        triggerBuild(participation);
+    }
+
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation, String commitHash, RepositoryType triggeredByPushTo) throws ContinuousIntegrationException {
+        log.warn("Triggering with a test push is not supported for Hades. Triggering build without test push.");
+        triggerBuild(participation, commitHash);
+    }
+
+    private void postJob(HadesBuildJobDTO job) {
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            headers.add(HttpHeaders.AUTHORIZATION, "Basic " + Base64.getEncoder().encodeToString(((hadesUsername + ":" + hadesPassword).getBytes())));
+            headers.add(HttpHeaders.CONTENT_TYPE, "application/json");
+            HttpEntity<HadesBuildJobDTO> request = new HttpEntity<>(job, headers);
+
+            var response = restTemplate.postForEntity(hadesServerUrl + "/build", request, JsonNode.class);
+            log.debug("Hades response: {}", response);
+        }
+        catch (Exception e) {
+            log.error("Error triggering the build for {}", job, e);
+        }
+    }
+
+    private HadesBuildJobDTO createJob(ProgrammingExerciseParticipation participation) throws JsonProcessingException {
+
+        // Build Job Metadata
+        var metadata = new HashMap<String, String>();
+        metadata.put("PLAN_KEY", participation.getBuildPlanId());
+
+        var steps = new ArrayList<HadesBuildStepDTO>();
+
+        // Create Clone Step
+        // TODO: We need a solution to clone with less critical credentials - This requires changes in the localvc implementation
+        var cloneMetadata = new HashMap<String, String>();
+        cloneMetadata.put("REPOSITORY_DIR", "/shared");
+        cloneMetadata.put("HADES_TEST_USERNAME", gitUsername);
+        cloneMetadata.put("HADES_TEST_PASSWORD", gitPassword);
+        cloneMetadata.put("HADES_TEST_URL", participation.getProgrammingExercise().getTestRepositoryUri());
+        cloneMetadata.put("HADES_TEST_PATH", "./");
+        cloneMetadata.put("HADES_TEST_ORDER", "1");
+        cloneMetadata.put("HADES_ASSIGNMENT_USERNAME", gitUsername);
+        cloneMetadata.put("HADES_ASSIGNMENT_PASSWORD", gitPassword);
+        cloneMetadata.put("HADES_ASSIGNMENT_URL", participation.getVcsRepositoryUri().toString());
+        cloneMetadata.put("HADES_ASSIGNMENT_PATH", "./assignment");
+        cloneMetadata.put("HADES_ASSIGNMENT_ORDER", "2");
```

Comment:
I think documenting some of these flags (especially stuff that is not immediately clear like "HADES_ASSIGNMENT_ORDER" would be helpful.
For maintainability it may also make sense to convert these into constants, but that is optional